src/wasm/asm-wasm-builder.cc

```diff
@@ -453,26 +574,108 @@ class AsmWasmBuilderImpl : public AstVisitor {
   }                                                                   \
   break
 
+  Expression* GetLeft(BinaryOperation* expr) {
```

Comment:
What I meant is, assuming that the "interesting" part of a type conversion like "x|0" is always on the left is not sound, and is likely to break on the very first "0|x". We need to be more careful here.